test/serializers/default_include_test.rb

```diff
@@ -0,0 +1,52 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
```

Comment:
Ah in this case I think my current test is not gonna work.
I'm assuming in this case that serialising the Blog we'll see the authors in the result, which are in the `default_include` of PostSerializer.

This is because I'm using `adapter: :attributes` in the test, which by default include first level associations (the equivalent of having `include: 'posts'` in JSON-API adapter).

Also, my implementation actually "merges" the controller include with the serialiser one. Are you saying the controller one should actually override the serialiser's?

Maybe a quick "expectations" diagram would help in this case 😅.
Let's take my example of Blog -> Posts (default_include :author) -> Author

1. What's the expected result for a simple `render json: Blog.first`?
2. What's the expected result for `render json: Blog.first, include: 'posts'`?
3. Let's assume posts have also another association "category", what's the expected result for `render json: Blog.first, include: 'posts.category'?